Makefile

```diff
@@ -22,14 +23,32 @@ gofmt:
 $(GOLINT):
 	go install golang.org/x/lint/golint
 
+$(GEN_ATOMICINT): $(wildcard ./internal/gen-atomicint/*)
+	go build -o $@ ./internal/gen-atomicint
+
 .PHONY: golint
 golint: $(GOLINT)
 	$(GOLINT) ./...
 
 .PHONY: lint
-lint: gofmt golint
+lint: gofmt golint generatenodirty
 
 .PHONY: cover
 cover:
 	go test -coverprofile=cover.out -coverpkg ./... -v ./...
 	go tool cover -html=cover.out -o cover.html
+
+.PHONY: generate
+generate: $(GEN_ATOMICINT)
+	go generate ./...
+
+.PHONY: generatenodirty
+generatenodirty:
+	@[ -z "$$(git status --porcelain)" ] || ( \
```

Comment:
why do we use `git status` here and `git diff --quiet` below? Can they both be the same?

Reply:
`git status` is more appropriate to check for uncommitted changes since `git diff` won't report staged changes by default.